crates/prover/src/components/poseidon.rs

```diff
@@ -0,0 +1,358 @@
+//! Component for Poseidon permutation rounds.
+//!
+//! # Columns
+//!
+//! - enabler
+//! - index
+//! - state (input)
+//! - inter_state (input + round constant)
+//! - inter_state_sq (inter_state * inter_state)
+//! - inter_state_quad (inter_state_sq * inter_state_sq)
+//! - s_box_out_state (inter_state * inter_state_quad)
+//! - full_round
+//! - final_round
+//!
+//! # Constraints
+//!
+//! * enabler is a bool
+//!   * `enabler * (1 - enabler)`
+//! * full_round is a bool
+//!   * `full_round * (1 - full_round)`
+//! * final_round is a bool
+//!   * `final_round * (1 - final_round)`
+//! * round constant addition
+//!   * `inter_state[i] = state[i] + round_constant[i]`
+//! * s-box computation (x^5)
+//!   * `inter_state_sq[i] = inter_state[i] * inter_state[i]`
+//!   * `inter_state_quad[i] = inter_state_sq[i] * inter_state_sq[i]`
+//!   * `s_box_out_state[i] = inter_state[i] * inter_state_quad[i]`
+//! * mds matrix multiplication
+//!   * `next_state = MDS * s_box_out_state`
+//! * poseidon round relation
+//!   * `+ [state[0], state[1]]` in `PoseidonRound` relation (input)
+//!   * `- [final_state[0]]` in `PoseidonRound` relation (output)
```

Comment:
more a convention, but all constraints are described directly as `expr` (=0), and uses columns, so not sure what `next_state` is